optimizely/marketing-website
# add set api domain function to test in staging

This PR addresses JIRA Issue, allowing QA of custom versions by assigning a custom apiDomain variable for all endpoints.
https://optimizely.atlassian.net/browse/MKT-1725

#### Overview

If on staging the `apiDomain` query param is present the global `window.apiDomain` variable will be replaced accordingly and all form action `apiDomain`'s will be replaced as well

The script for this logic is only injected into the head if the `staging` grunt task is run.

#### Additional

- Cleaned up scripts in the head to be uglified with `scriptInjector` handlebars helper
- Moved all `<head>` scripts into their own directory

#### To Test

Go to https://www.optimizelystaging.com/dfoxpowell/custom-version-api-domain/ and load the page. See that there are no errors and that you can submit forms successfully. Also in the console check that `window.apiDomain` is `//app.optimizely.com` and that form actions also include this apiDomain

Next go to https://www.optimizelystaging.com/dfoxpowell/custom-version-api-domain/?apiDomain=something. Check that `window.apiDomain` is `something` and that form action strings now include `something`.

To test the production build run `grunt build` and check the head to see that this script IS NOT being injected into the head
